src/groupeddataframe/groupeddataframe.jl

```diff
@@ -33,6 +33,190 @@ mutable struct GroupedDataFrame{T<:AbstractDataFrame}
                                          # thread safe
 end
 
+"""
+    groupby(d::AbstractDataFrame, cols; sort=false, skipmissing=false)
+
+Return a `GroupedDataFrame` representing a view of an `AbstractDataFrame` split
+into row groups.
+
+# Arguments
+- `df` : an `AbstractDataFrame` to split
+- `cols` : data frame columns to group by. Can be any column selector
+  ($COLUMNINDEX_STR; $MULTICOLUMNINDEX_STR).
+- `sort` : whether to sort groups according to the values of the grouping columns
+  `cols`; if `sort=false` then the order of groups follows the order of appereance
```

Comment:
This phrasing implies we commit to keeping this order at least until 2.x. I think we should say that this is the current implementation but stress that it is undefined. Indeed it's possible that radix sort-based implementations could be faster  in some cases and it would be nice not to close that possibility.

Also:
```suggestion
  `cols`; if `sort=false` then the order of groups follows the order of appearance
```

Reply:
OK - added a passage that it is undefined